airbyte-integrations/connectors/destination-mongodb/src/main/resources/spec.json

```diff
@@ -8,24 +8,79 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "MongoDB Destination Spec",
     "type": "object",
-    "required": ["host", "port", "database", "auth_type"],
+    "required": ["database", "auth_type"],
     "additionalProperties": true,
     "properties": {
-      "host": {
-        "title": "Host",
-        "description": "Hostname of the database.",
-        "type": "string",
-        "order": 0
-      },
-      "port": {
-        "title": "Port",
-        "description": "Port of the database.",
-        "type": "integer",
-        "minimum": 0,
-        "maximum": 65536,
-        "default": 27017,
-        "examples": ["27017"],
-        "order": 1
+      "instance_type": {
+        "type": "object",
+        "title": "MongoDb instance type",
+        "description": "MongoDb instance to connect to. For MongoDB Atlas and Replica Set TLS connection is used by default.",
+        "order": 0,
+        "oneOf": [
```

Comment:
Can you follow the `oneOf` pattern specified in the following doc?
https://docs.airbyte.io/connector-development/connector-specification-reference#using-oneofs

Basically we want to add a common field for all the `oneOf` members so that it is easy to tell which member is selected by the consumer (e.g. in the `getConnectionString` method).

An example can be found here:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json#L101

Each `compression_codec` member has the `codec` property. When parsing the `compression_codec`, we only need to check this property to construct the config:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/S3AvroFormatConfig.java#L35

Reply:
Thanks a lot, I made changes, it really looks better and is much more convenient.